### PR TITLE
Refactored FindView

### LINKED ISSUE #531: Refactor FindView
### Description
Refactor FindView.vue and related components because :star: future-proofing :star:!

### Acceptance Criteria
- [x] Findview.vue is refactored
- [x] BookListComponent is refactored
- [x] LoanListComponent is refactored
- [ ] UserListComponent is refactored

### PR DESCRIPTION
**Pull request by te4-alexander-kjellberg**
## ⚙️ What issue did you implement or fix?
Fixes #531 

## 📜 Summary
- Refactored FindView.vue
- Refactored UserListComponent.vue

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/views/FindView.vue [hunk 1]
```diff
@@ -71,24 +70,34 @@ import UsersModule from '@/store/modules/UsersModule';
...
   },
-})
-export default class FindView extends Vue {
-  private RoleChecker: RoleChecker = RoleChecker;
-  users:boolean = false;
-  loans:boolean = false;
-  books:boolean = false;
+  setup() {
+    let rolechecker: RoleChecker = RoleChecker;
+    let users = ref(false as boolean);
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/views/FindView.vue:83]: Do you really need to specify that this is a boolean?

Comment (te4-adrian-almetun-smeds) [reply frontend/src/views/FindView.vue:83]: These `ref`s could also use `cons` instead of `let`

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Looks good. Just got a small comment

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
:+1: Looks all good.

**Review by TE4-Rasmus-Brednert — APPROVED** [review]
Well done!

**Pull request merged**